aws/resource_aws_iam_role_test.go

```diff
@@ -51,18 +51,36 @@ func testSweepIamRoles(region string) error {
 	}
 	conn := client.(*AWSClient).iamconn
 	prefixes := []string{
+		"batch_tf_acc_test",
+		"codepipeline-",
+		"cognito_authenticated_",
+		"cognito_unauthenticated_",
 		"ecs_instance_role",
 		"ecs_tf",
 		"EMR_AutoScaling_DefaultRole_",
+		"enhanced-monitoring-role-",
+		"es-domain-role-",
+		"foo_role",
+		"foo-role",
 		"iam_emr",
+		"iam_for_lambda",
+		"iam_for_sfn",
+		"ssm_role",
 		"terraform-",
 		"test",
 		"tf",
 	}
```

Comment:
Also noted these prefixes in role `name` attributes across tests:
* `foobar` : L143 & L233 of resource_aws_autoscaling_lifecycle_hook_test.go
* `event_` : L855 & L1014 of resource_aws_cloudwatch_event_target_test.go 
* `CWLtoKinesisRole_` : L109 of resource_aws_cloudwatch_log_destination_policy_test.go and L109 of  resource_aws_cloudwatch_log_destination_test.go
* `rds` : (rds_backup) L748 of resource_aws_db_option_group_test.go, (rds_sample_role) L2607, 2700 of resource_aws_rds_cluster_test.go
* `another_rds` : L2644, 2737, 2794 of resource_aws_rds_cluster_test.go
* `role` : L125-271 of resource_aws_iot_role_alias_test.go
* `firehose` : L1072 of resource_aws_msk_cluster_test.go
* `sns-delivery-status` : L686 of resource_aws_sns_topic_test.go
* `ssm-role` : L470 of resource_aws_ssm_maintenance_window_task_test.go

Reply:
Phew! We really need to wrangle these in. 😅  Thank you so much for going through and finding these, I'll add them to a technical debt issue to standardize and add them to this list.

Reply:
Issue created: https://github.com/terraform-providers/terraform-provider-aws/issues/14003